DeployJenkinsfile

```diff
@@ -15,161 +16,239 @@ pipeline {
         buildDiscarder(logRotator(artifactNumToKeepStr: '30', numToKeepStr: '30'))
     }
     parameters  {
-        choice(name: 'DEPLOYMENT_TYPE', choices:'cf3-release-candidate\ncf3-staging\ncf3-integration\nvpc\neu-central\ngovw\nperf-cf3', description: 'This specifies which point of presence to deploy to')
+        choice(name: 'DEPLOYMENT_TYPE', choices:'cf3-release-candidate\ncf3-staging\ncf3-integration\nvpc\neu-central\ngovw\nperf-cf3\nint\nstage\nqa', description: 'This specifies which point of presence to deploy to')
+        booleanParam(name: 'DEPLOY_CF', defaultValue: false, description: 'Determines whether to deploy changes to CF environment.')
+
         booleanParam(name: 'MAP_ROUTES_TO_DEPLOYED_APP', defaultValue: true, description: 'Determines if the published routes are mapped to the newly deployed app. Defaults to true.')
         booleanParam(name: 'RUN_FULL_ACCEPTANCE_TESTS', defaultValue: true, description: 'Determines whether to run the full acceptance test suite against the freshly deployed UAA. If left unchecked, it will run at least the basic acceptance tests.')
         string(name: 'UAA_CI_CONFIG_BRANCH', defaultValue: 'master',
                 description: 'uaa-cf-release repo branch to use for testing/deployment')
+        booleanParam(name: 'DEPLOY_EKS', defaultValue: false, description: 'Determines whether to update GitOps repository to deploy changes to EKS.')
+        string(name: 'IMAGE_TAG', defaultValue: 'iam-uaa:75.18.1-146',
+                description: 'Grid Artifactory image tag pushed from Build Pipeline')        
     }
-    stages{
-        stage('Deploy') {
-            agent{
-                docker {
-                    image "${NODE['IMAGE']}"
-                    label "${NODE['LABEL']}"
-                    args "${NODE['ARGS']}"
-                }
-            }
-            environment {
-                CF_CREDENTIALS = credentials("CF_CREDENTIALS_${DEPLOYMENT_TYPE.toUpperCase().replaceAll('-','_')}")
-                ADMIN_CLIENT_SECRET = credentials("ADMIN_CLIENT_SECRET_${DEPLOYMENT_TYPE.toUpperCase().replaceAll('-','_')}")
-                SPLUNK_OTEL_ACCESS_TOKEN = credentials("SPLUNK_OTEL_ACCESS_TOKEN")
+    stages
+    {
+        stage('CF Deploy')
+        {
+            when 
+            {
+                expression { params.DEPLOY_CF == true }
```

Comment:
Specify `beforeAgent true` directive within `when` condition block; otherwise, pipeline goes into `CF Deploy` stage and loads credentials. Some credentials like ADMIN_CLIENT_SECRET are looked up based on CF deploy environment. If deploying to EKS and selecting `int` as deployment environment, these credentials won't be found and pipeline will fail.  CF stages execution will be skipped afterwards, but I guess loading credentials is part of agent start up.

Anyway, you could take a look at [rc_75.18.3/42/pipeline](https://i.ci.build.ge.com/bjmjek1k/ci/blue/organizations/jenkins/UAA%2FDeploy/detail/rc_75.18.3/42/pipeline) vs. [rc_75.18.3/43/pipeline](https://i.ci.build.ge.com/bjmjek1k/ci/blue/organizations/jenkins/UAA%2FDeploy/detail/rc_75.18.3/43/pipeline/) where I added beforeAgent directive